lib/actions/push.ts

```diff
@@ -98,30 +113,53 @@ async function getAppOwner(sdk: BalenaSDK, appName: string) {
 	return selected.extra;
 }
 
-export const push: CommandDefinition<
-	{
-		applicationOrDevice: string;
-	},
-	{
-		source: string;
-		emulated: boolean;
-		nocache: boolean;
+function parseRegistrySecrets(secretsFilename: string): RegistrySecrets {
+	let registrySecrets: RegistrySecrets = {};
+	if (secretsFilename) {
+		let isYaml = false;
+		if (/.+\.ya?ml$/i.test(secretsFilename)) isYaml = true;
+		else if (!/.+\.json$/i.test(secretsFilename))
+			throw new Error(
+				'The registry-secrets filename must end with .json, .yml or .yaml',
+			);
+
+		const raw = require('fs')
+			.readFileSync(secretsFilename)
```

Comment:
We shouldn't be doing synchronous file I/O. You can do
```
import { fs } from 'mz';
```

To get a nice promise based interface to fs, and using `async/await` there would also be preferred.

Reply:
Done as suggested, thanks. In this particular case it wasn't much of an advantage -- I think the CLI has nothing else to do while waiting for the file to be read asynchronously. I imagine that the async (hidden promise) handling makes the code run marginally slower.